File: api/src/main/java/no/difi/meldingsutveksling/api/NextMoveQueue.java
package no.difi.meldingsutveksling.api;

import no.difi.meldingsutveksling.ServiceIdentifier;
import no.difi.meldingsutveksling.domain.sbdh.StandardBusinessDocument;
import org.springframework.core.io.Resource;

public interface NextMoveQueue {

    /**
     * Enqueue an incoming nextmove message. [asic] is closed.
     */
    void enqueueIncomingMessage(StandardBusinessDocument sbd,ServiceIdentifier serviceIdentifier, Resource asic);

    /**
     * Enqueue an incoming message.
     */
    void enqueueIncomingMessage(StandardBusinessDocument sbd, ServiceIdentifier serviceIdentifier);

}

File: api/src/main/kotlin/no/difi/meldingsutveksling/api/NextMoveQueue.kt
package no.difi.meldingsutveksling.api

import no.difi.meldingsutveksling.ServiceIdentifier
import no.difi.meldingsutveksling.domain.sbdh.StandardBusinessDocument
import org.springframework.core.io.Resource

interface NextMoveQueue {

    /**
     * Enqueue an incoming nextmove message. [asic] is closed.
     */
    fun enqueueIncomingMessage(sbd: StandardBusinessDocument, serviceIdentifier: ServiceIdentifier, asic: Resource?)

    /**
     * Enqueue an incoming message.
     */
    fun enqueueIncomingMessage(sbd: StandardBusinessDocument, serviceIdentifier: ServiceIdentifier)
}